### DISCUSSION TITLE
Roadmap Update - November 2024

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#13991] [Announcements]
# Roadmap Update - November 2024

Hello Immich fans, as the holiday season approaches and the year all-to-quickly draws to a close, we have received a few questions regarding our [public roadmap](https://immich.app/roadmap) that we would like to clarify at this time.

For many, the most important item on the list is the "Stable release". Prior to a stable release, we have a list of tasks that we are actively working on. Those include the following:

0. Mobile-server synchronization rewrite
1. Mobile database migration (Isar to SQLite)
2. Mobile backward compatibility strategy
3. Server migration from TypeORM to Kysely
4. Change the default `UPLOAD_LOCATION` from `/usr/src/app/upload` to `/data`
5. A new asset visibility field (something like `{ visibility: 'hidden' | 'archived' | 'private' }`)
6. Svelte 5 migration

With that said, we are now expecting to have a stable release in early 2025. As a result, we have deprioritized other features, which are now targeted for sometime next year. Below is the updated roadmap.

![updated-roadmap](https://github.com/user-attachments/assets/91e447ed-0b8b-4c22-a2de-ff2989a307a9)

We are excited about these changes and hope you can be patient with us as we continue to work towards them. Our goal is to make an app that delights and can be used without sacrificing your privacy or fear of platform lock-in.

Thank you, again, for your support and love for the project. We are very happy to see how much the community and Immich's adoption have grown throughout the year. 

Cheers!

Immich team

**Comment by HELPY4** [comment]
Good! 📈

**Comment by MmDawN** [comment]
Looking forward to the stable version and stable mobile app!

**Comment by Etienne-bdt** [comment]
Hey there, I am really curious about the editing features, I even considered forking the repo to work on some ! Will the features be included on the main server or in the respective apps ?

**Reply by alextran1502** [reply]
The mobile app already has some basic editing feature such as crop/rotate/filter. The feature in the road map would include the web app with perhaps the ability to perform server side editing

**Comment by oxodao** [comment]
Nice this looks quite promising! Really looking forward to these features 😀

When adding a media to an album on desktop, you have a search bar to filter your albums and find the correct one. I don't see this on mobile, is there any plan to add this ? It's a bit painful especially because the albums are not alphabetically sorted..

**Comment by jrstrunk** [comment]
Take your time to make things right before going stable, this is all really exciting!

**Comment by therealhoodboy** [comment]
Very much looking forward to test this app on my Raspi 5!

**Reply by LoggeL** [reply]
Running it myself on one. Sailing smoothly

**Reply by therealhoodboy** [reply]
Yes can agree, it works very good. I am impressed how smoothly it can replace iCloud Photos, just awesome.

**Comment by SadeQuin50** [comment]
Please help me

**Comment by SadeQuin50** [comment]
What this

**Reply by therealhoodboy** [reply]
I would say a very good software to host your own state-of-the-art cloud photo service.

**Comment by athornfam2** [comment]
Does anyone have an add-on to immich to compress photos? I don't need the raw output of photos.

**Reply by milos7250** [reply]
If you find one let me know 😀. I think it would be an interesting feature to be able to selectively choose quality of images to be stored.

**Reply by milos7250** [reply]
Short feature request browsing session later, I've found that the developers have no intention of implementing this and marked it "out of scope" for the project. Which is a shame in my opinion. There is a proxy that you can add in between https://gist.github.com/JamesCullum/6604e504318dd326a507108f59ca7dcd that compresses all uploads, so it does not really give you an option to retain some images in original quality. Ideally, I'd like to be able to tag certain images for "small size" option and have the server to compress these server-side

**Reply by athornfam2** [reply]
I was thinking of writing a shell script myself to resize the files... something like that as a system task next to transcoding would be nice. I only need 80% of quality to reduce the size of the data store for these photos and videos. 28,000 photos and 1200 videos at 175GB's seems a little high to me which doesn't include the 75GB's for thumbnails and everything else just to show the photo or video.

**Comment by freak12techno** [comment]
Is it possible to prioritize the in-box backup/restore solution (as in, a way to do a database + assets backup from the Immich UI, then be able to download it and restore via Immich UI later, and ideally be able to set it up to upload backups to S3/FTP/whatever periodically) at some point after the stable release?
I assume that especially during the database migrations stuff can change rapidly, so there's little point of working on it before it's stable, but this would be quite important to have IMO, and again IMO this should have more priority than most of the other features.

For context, would link my message in a different thread on backup/restore on how it might look like: https://github.com/immich-app/immich/discussions/5441#discussioncomment-9899387

**Reply by bo0tzz** [reply]
> IMO this should have more priority than most of the other features.

Ask 10 Immich users which feature should have the highest priority, and you'll get 15 answers.

**Reply by freak12techno** [reply]
@bo0tzz I agree, that's why IMO part.
But for me I'd rather have the system that lacks some features but has the tools to easily backup/restore it in case of a malfunction than have a system with a lot of fancy features but I have to take care of the backups myself.
